APMrover2/RC_Channel.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <RC_Channel/RC_Channel.h>
+#include "Rover.h"
+#include "mode.h"
+
+class RC_Channel_Rover : public RC_Channel
+{
+
+public:
+
+protected:
+
+    void init_aux_function(aux_func_t ch_option, aux_switch_pos_t) override;
+    void do_aux_function(aux_func_t ch_option, aux_switch_pos_t) override;
+
+    // called when the mode switch changes position:
+    void do_mode_switch_mode_change(modeswitch_pos_t new_pos,
+                                    modeswitch_pos_t prev_pos) override;
+
+private:
+
+    void do_aux_function_change_mode(Mode &mode,
+                                     const aux_switch_pos_t ch_flag);
+
+};
+
+class RC_Channels_Rover : public RC_Channels
```

Comment:
I think this whole class can be left as a part of the RC_Channel library without vehicles needing to override it.